text/0011-html_viz.md

```diff
@@ -0,0 +1,182 @@
+
+While care will be taken to keep `index.html` files small, users may understandably want to disable them entirely. To achieve this we'll add a new flag to `qri save` and `qri update`: `--no-render`. No render will prevent the execution of any viz template. This will save a few KB from version to version at the cost of usability.
+
+Users can _override_ the default template by supplying their own custom viz templates either by specifying a `viz.scriptPath`:
+
+`dataset.yaml`:
+```yaml
+name: example_dataset
+# additional fields elided ...
+viz:
+  syntax: html
+  scriptPath: template.html
+```
+
+and running save:
+```
+$ qri save --file dataset.yaml
+```
+
+Or by running `qri save` with an `.html` file argument:
+`qri save --file template.html me/example_dataset`
+
+Since the above example provided no additional configuration details for the `viz` component in `dataset.yaml`, the two calls will have the same effect.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Template API
+
+Introducing HTML template execution requires defining an API for template values. This API will need to be documented & maintained just like any other API in the Qri ecosystem.
+
+The template implementation will use the [go language html template package](https://golang.org/pkg/html/template)
+
+#### Dataset is exposed as `ds`
+
+HTML template should expose a dataset document at `ds`. By exposing the document as `ds`, it matches our for referring to a dataset in starlark., and allows access to all defined parts of a dataset. ds should use _lowercase_ fields for component accessors. eg:
+
+```html
+{{ ds.meta.title }}
+```
+
+Undefined components should be defined as empty struct pointers if null. For example a dataset _without_ a transform the following template should print nothing:
+```html
+{{ ds.transform }}
+```
+And this should error:
+```html
+{{ ds.transform.script }}
+```
+
+Having default empty pointers prevents unnecessary `if` clauses, allowing a skip to tests for existence on component fields:
+```html
+{{ if ds.transform.script }}
+{{ end }}
+```
+
+### Template functions
+
+Top level functions should be loaded into the template `funcs` space to make rendering templates easier. The go html template package comes with [predefined functions](https://golang.org/pkg/text/template/#hdr-Functions). Because our implementation builds the html/template package, this RFC introduces all of these functions into our template engine. I think this is totally fine, might be a pain if someone needs to write a non-go `Render` implementation, but at least the .
```

Comment:
"Because our implementation builds the" -> "Because our implementation builds *on* the"

Sentence is incomplete at the end of this paragraph